Rubberduck.VBEEditor/Events/VbeEvents.cs

```diff
@@ -227,51 +232,34 @@ private void ProjectReferenceRemovedHandler(object sender, ReferenceEventArgs e)
 
         public event EventHandler EventsTerminated;
 
-        #region IDisposable
+        private long _terminated;
+        private const long _true = 1;
+        public bool Terminated => Interlocked.Read(ref _terminated) == _true;
 
-        private bool _disposed;
-        /// <remarks>
-        /// This is a not a true implementation of IDisposable pattern
-        /// because the method is made private and is available only
-        /// via the static method <see cref="Terminate"/> to provide
-        /// a single point of entry for disposing the singleton class
-        /// </remarks>
-        private void Dispose(bool disposing)
+        private void TerminateInternal()
         {
-            if (!_disposed && _projects != null)
-            {
-                EventsTerminated?.Invoke(this, EventArgs.Empty);
+            // If we fail, we at least should advertise that we're now dead
+            Interlocked.Exchange(ref _terminated, _true);
 
-                var projectIds = _components.Keys.ToArray();
-                foreach (var projectid in projectIds)
-                {
-                    UnregisterProjectHandlers(projectid);
-                }
-                
-                _projects.ProjectActivated -= ProjectActivatedHandler;
-                _projects.ProjectRenamed -= ProjectRenamedHandler;
-                _projects.ProjectRemoved -= ProjectRemovedHandler;
-                _projects.ProjectAdded -= ProjectAddedHandler;
-                _projects.DetachEvents();
-                _projects.Dispose();
-                
-                _disposed = true;
+            EventsTerminated?.Invoke(this, EventArgs.Empty);
+            EventsTerminated = delegate { };
+            var projectIds = _components.Keys.ToArray();
+            foreach (var projectid in projectIds)
+            {
+                UnregisterProjectHandlers(projectid);
             }
-        }
 
-        ~VBEEvents()
-        {
-            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-            Dispose(false);
-        }
+            if (_projects == null)
```

Comment:
See above. The only way this will ever be null is if something kills it after the ctor runs.  Should probably be something more like ``if(_projects?.IsWrappingNullReference ?? true)`